app/models/semester_registration.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class SemesterRegistration < ApplicationRecord
+  # callbacks
+  before_validation :assign_academic_term
+  before_validation :assign_semester
+
+  # enums
+  enum status: { draft: 0, saved: 1 }
+
+  # relations
+  belongs_to :academic_term
+  belongs_to :student
+  has_many :course_enrollments, dependent: :destroy
+
+  # validations
+  validates :semester, uniqueness: { scope: :student_id }
+  validates :semester, numericality: { greater_than: 0 }
```

Comment:
`validates :semester, uniqueness: { scope: :student_id }, numericality: { greater_than: 0 }` aynı attribute için validasyonları aynı satırda yazalım.  
https://doc.omu.sh/policy/rails/